huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go

```diff
@@ -0,0 +1,151 @@
+package ddm
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getDdmAccountResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getAccount: Query DDM account
+	var (
+		getAccountHttpUrl = "v1/{project_id}/instances/{instance_id}/users"
+		getAccountProduct = "ddm"
+	)
+	getAccountClient, err := cfg.NewServiceClient(getAccountProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DDM Client: %s", err)
+	}
+
+	parts := strings.SplitN(state.Primary.ID, "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid id format, must be <instance_id>/<account_name>")
+	}
+	instanceID := parts[0]
+	accountName := parts[1]
+	getAccountPath := getAccountClient.Endpoint + getAccountHttpUrl
+	getAccountPath = strings.ReplaceAll(getAccountPath, "{project_id}", getAccountClient.ProjectID)
+	getAccountPath = strings.ReplaceAll(getAccountPath, "{instance_id}", fmt.Sprintf("%v", instanceID))
+
+	getAccountResp, err := pagination.ListAllItems(
+		getAccountClient,
+		"offset",
+		getAccountPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DdmAccount: %s", err)
+	}
+
+	getAccountRespJson, err := json.Marshal(getAccountResp)
+	if err != nil {
+		return nil, err
+	}
+	var getAccountRespBody interface{}
+	err = json.Unmarshal(getAccountRespJson, &getAccountRespBody)
+	if err != nil {
+		return nil, err
+	}
+
+	accounts := utils.PathSearch("users", getAccountRespBody, nil)
+	if accounts == nil {
+		return nil, fmt.Errorf("the instance %s has no account", instanceID)
+	}
+	for _, account := range accounts.([]interface{}) {
+		name := utils.PathSearch("name", account, nil)
+		if accountName != name {
+			continue
+		}
+		return account, nil
+	}
+
+	return nil, fmt.Errorf("the instance %s has no account", instanceID)
```

Comment:
Make the error message clear.
`return nil, fmt.Errorf("the instance %s has no account %s", instanceID, accountName)`

Reply:
done